core/templates/pages/blog-dashboard-page/blog-post-editor/blog-post-editor.component.ts

```diff
@@ -0,0 +1,314 @@
+  windowIsNarrow: boolean;
+  defaultTagsList: string[];
+  maxAllowedTags: number;
+  contentEditorIsActive: boolean = true;
+  allowedImageFormats: readonly string[] = AppConstants.ALLOWED_IMAGE_FORMATS;
+  filename: string;
+  localEdittedContent: string;
+  thumbnailDataUrl: string;
+  invalidImageWarningIsShown: boolean = false;
+  newChangesAreMade: boolean = false;
+  latestChangesarePublished: boolean = false;
+  MAX_CHARS_IN_BLOG_POST_TITLE: number;
+  HTML_SCHEMA: EditorSchema = {
+    type: 'html',
+    ui_config: {
+      hide_complex_extensions: false
+    }
+  };
+
+  constructor(
+    private alertsService: AlertsService,
+    private blogPostEditorBackendService: BlogPostEditorBackendApiService,
+    private loaderService: LoaderService,
+    private urlInterpolationService: UrlInterpolationService,
+    private blogDashboardPageService: BlogDashboardPageService,
+    private ngbModal: NgbModal,
+    private blogPostUpdateService: BlogPostUpdateService,
+    private changeDetectorRef: ChangeDetectorRef,
+    private contextService: ContextService,
+    private imageLocalStorageService: ImageLocalStorageService,
+    private assetsBackendApiService: AssetsBackendApiService,
+    private windowDimensionService: WindowDimensionsService,
+  ) {}
+
+  ngOnInit(): void {
+    this.loaderService.showLoadingScreen('Loading');
+    this.DEFAULT_PROFILE_PICTURE_URL = this.urlInterpolationService
+      .getStaticImageUrl('/general/no_profile_picture.png');
+    this.blogPostData = BlogPostData.createInterstitialBlogPost();
+    this.blogPostId = this.blogDashboardPageService.blogPostId;
+    this.title = this.blogPostData.title;
+    this.MAX_CHARS_IN_BLOG_POST_TITLE = (
+      AppConstants.MAX_CHARS_IN_BLOG_POST_TITLE);
+    this.loaderService.hideLoadingScreen();
+    this.initEditor();
+    this.contextService.setImageSaveDestinationToLocalStorage();
+    this.windowIsNarrow = this.windowDimensionService.isWindowNarrow();
+    this.windowDimensionService.getResizeEvent().subscribe(() => {
+      this.windowIsNarrow = this.windowDimensionService.isWindowNarrow();
+    });
+  }
+
+  getSchema(): EditorSchema {
+    return this.HTML_SCHEMA;
+  }
+
+  initEditor(): void {
+    this.blogPostEditorBackendService.fetchBlogPostEditorData(this.blogPostId)
+      .then(
+        (editorData: BlogPostEditorData) => {
+          this.blogPostData = editorData.blogPostDict;
+          this.authorProfilePictureUrl = decodeURIComponent((
+            // eslint-disable-next-line max-len
+            editorData.profilePictureDataUrl || this.DEFAULT_PROFILE_PICTURE_URL));
+          this.authorUsername = editorData.username;
+          this.defaultTagsList = editorData.listOfDefaulTags;
+          this.maxAllowedTags = editorData.maxNumOfTags;
+          this.title = this.blogPostData.title;
+          this.dateTimeLastSaved = this.getDateStringInWords(
+            this.blogPostData.lastUpdated);
+          if (this.blogPostData.content !== '') {
```

Comment:
```suggestion
          if (this.blogPostData.content.length > 0) {
```

Reply:
Done.

Reply:
Please fix this everywhere else too. [Example](https://github.com/oppia/oppia/pull/13522/files#diff-a291a1535e1b49bda06a26242b423008dfa67660b0bea27fc3d1fab60f15a865R159).

Reply:
Done.